openmc/model/model.py

```diff
@@ -324,6 +325,26 @@ def init_lib(self, threads=None, geometry_debug=False, restart_file=None,
         # communicator
         openmc.lib.init(args=args, intracomm=intracomm, output=output)
 
+    def sync_dagmc_universes(self):
+        """
+        Synchronize all DAGMC universes in the current geometry.
+        This method iterates over all DAGMC universes in the geometry and
+        synchronizes their cells with the current material assignments.
+
+        .. versionadded:: 0.15.1-dev
+
+        """
+        if self.is_initialized:
+            if self.materials:
+                materials = self.materials
+            else:
+                materials = self.geometry.get_all_materials().values()
+            for dagmc_universe in self.geometry.get_all_dagmc_universes().values():
+                dagmc_universe.sync_dagmc_cells(materials)
```

Comment:
This is the only place that `get_all_dagmc_universes` is called. I wonder if doing the DAGMCUniverse check here wouldn't suffice.

```suggestion
            for univ in self.geometry.get_all_universes().values():
                if isinstance(univ, openmc.DAGMCUniverse):
                    univ.sync_dagmc_cells(materials)
```